lib/tasks/delete_old_targets.rake

```diff
@@ -0,0 +1,5 @@
+task :delete_old_targets => :environment do
+  Target.where('created_at < ?', 1.week.ago).each do |target|
```

Comment:
Agree with moving the where to a scope.
Also, maybe you can use `find_each` instead of just `each`, it's a bit more performant when you have large numbers since it does not load all the records in memory at once.